README.md

```diff
@@ -13,121 +13,274 @@ and the Flutter guide for
-    <script>
-        const AUTH_DESTINATION_KEY = "openidconnect_auth_destination_url";
-        const AUTH_RESPONSE_KEY = "openidconnect_auth_response_info";
-
-        window.onload = function () {
-            if (window.opener && window.opener !== window) { //Used when working as a popup. Uses post message to respond to the parent window                
-                var parent = window.opener ?? window.parent;
-                parent.postMessage(location.href, "*");
-            } else { //Used for redirect loop functionality.                
-                //Get the original page destination                
-                const destination = sessionStorage.getItem(AUTH_DESTINATION_KEY || "/");
-                sessionStorage.removeItem(AUTH_DESTINATION_KEY);
-                //Store the current window location that will be used to get the information for authentication
-                sessionStorage.setItem(AUTH_RESPONSE_KEY, window.location);
-
-                //Redirect to where we're going so that we can restore state completely
-                location.assign(destination);
-            }
-        }
-    </script>
-</head>
-
-<body>
-</body>
-
-</html>
-```
+An example project that demonstrates `solid_auth` usage can be found [here](https://github.com/anusii/solid_auth/tree/main/example).
+
+## 🙏 Acknowledgments
+
+This library builds upon the excellent work of the [Bdaya-Dev/oidc](https://github.com/Bdaya-Dev/oidc) team. We are standing on the shoulders of giants! 
+
+Special thanks to:
+- **[Bdaya-Dev/oidc](https://pub.dev/packages/oidc)** - The robust, well-maintained OpenID Connect implementation that powers this library
+- **[oidc_default_store](https://pub.dev/packages/oidc_default_store)** - Secure, platform-appropriate token storage
+- The broader Solid and OpenID Connect communities for their specifications and guidance
+
+The solid_auth library focuses specifically on Solid pod authentication while leveraging these excellent foundational libraries for the core OIDC functionality.
+
+## 🔗 Links
+
+- [Solid Project](https://solidproject.org/)
+- [Solid OIDC Specification](https://solid.github.io/solid-oidc/)  
+- [WebID Specification](https://www.w3.org/2005/Incubator/webid/spec/identity/)
+- [Example Application](https://github.com/anusii/solid_auth/tree/main/example)
+- [Issue Tracker](https://github.com/anusii/solid_auth/issues)
+
+---
+
+## Roadmap
+
+### Offline-First Support
+Currently, `solid_auth` requires network connectivity during initialization to:
+- Discover identity providers from WebID profiles
+- Fetch OIDC provider configurations
+- Validate authentication sessions
+
+**Future Goal**: Enable fully offline-first applications that can start and function without network connectivity, using cached authentication data and provider configurations.
+
+This is essential for truly offline-capable Solid applications, but requires careful consideration of security trade-offs and cache management strategies.
+
```

Comment:
I saw, that there currently is another limitation: apparently, token refresh is not fully working. Leaving the tab of my solid_task app open, after some time (an hour or so) I saw synchronization errors due to the access token not being available. Starting the app today shows me that I am logged out again - so the persistence of the login data seems to be limited. Do you think we should document this here?

Reply:
I found the reason - I have to send a certain promt, which I now do. So this is now fixed.